flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableOutputFormat.java

```diff
@@ -259,13 +259,13 @@ public void configure(Configuration parameters) {
 	public void open(int taskNumber, int numTasks) throws IOException {
 		try {
 			StorageDescriptor sd = hiveTablePartition.getStorageDescriptor();
-			serializer = (Serializer) Class.forName(sd.getSerdeInfo().getSerializationLib()).newInstance();
-			Preconditions.checkArgument(serializer instanceof Deserializer,
-					"Expect to get a SerDe, but actually got " + serializer.getClass().getName());
-			ReflectionUtils.setConf(serializer, jobConf);
+			recordSerDe = (Serializer) Class.forName(sd.getSerdeInfo().getSerializationLib()).newInstance();
```

Comment:
Can we assert the type to be "Serializer" before casting it, similar to below for "Deserializer"?